packages/workflow/src/RoutingNode.ts

```diff
@@ -564,6 +569,102 @@ export class RoutingNode {
 						responseData.push(...tempResponseData);
 					} while (tempResponseData.length && tempResponseData.length === properties.pageSize);
 				}
+				// Generic pagination, using the contents of the previous response to
+				// decide whether a new one should be made.
+				else if (requestOperations.pagination.type === 'generic') {
+					const { properties } = requestOperations.pagination;
+
+					// Merge request properties coming in from different parts of the declaration
+					// (avoiding Object.assign() as it only does a shallow merge)
+					if (properties.request) {
+						requestData.options = merge({}, requestData.options, properties.request);
+					}
+
+					let previousResponseData: INodeExecutionData | null = null;
+					let currentResponseData: INodeExecutionData[] = [];
+					let canContinue = false;
+
+					do {
+						// Deep-clone requestData, as evaluating a parameter would overwrite it
+						const currentRequestData: DeclarativeRestApiSettings.ResultOptions = JSON.parse(
+							JSON.stringify(requestData),
+						);
```

Comment:
Didn't get to the bottom of why evaluating the requestData would overwrite the original request. However, we need to have the non-evaluated value available, to be able to re-evaluate against the response for every results page anew.

Reply:
Makes sense - you can use `cloneDeep` from lodash as well.

Reply:
I think I'll actually do just that. We're generally using the `JSON.parse(JSON.stringify())` pattern a bit too much for my taste, and could abstract away from it!